tests/common/test_validation.py

```diff
@@ -241,3 +242,14 @@ def test_nested_union(test_doc: TTestRecord) -> None:
         validate_dict(TTestRecord, test_doc, ".")
     assert e.value.field == "f_optional_union"
     assert e.value.value == "blah"
+
+
+def test_no_name() -> None:
+    class TTestRecordNoNoame(TypedDict):
+        name: TTableHintTemplate[str]
+
+    test_item = {"name": "test"}
```

Comment:
@IlyaFaer could you please also add a test for callable, e.g. if instead of string "test" here, there's a lambda:

```
TDynHintType = TypeVar("TDynHintType")
TFunHintTemplate = Callable[[TDataItem], TDynHintType]
TTableHintTemplate = Union[TDynHintType, TFunHintTemplate[TDynHintType]]
```